mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -394,8 +378,14 @@ public void close() throws IOException {
               String.format("Cannot read %s file: %s", file.format().name(), file.path()));
       }
       currentCloseable = iterable;
-      //TODO: Apply residual filtering before returning the iterator
-      return iterable.iterator();
+      boolean applyResidualFiltering = !context.getConfiguration().getBoolean(SKIP_RESIDUAL_FILTERING, false);
+      if (applyResidualFiltering && currentTask.residual() != null &&
```

Comment:
Can you assign the result of currentTask.residual() to a local variable to reduce method call?

Reply:
I agree with this. And we can also shorten the boolean variable name to `applyResidual`.

Reply:
OK, will change this.